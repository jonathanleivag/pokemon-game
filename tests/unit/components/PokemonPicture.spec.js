import { shallowMount } from '@vue/test-utils'
import PokemonPicture from '@/components/PokemonPicture.vue'

describe('PokemonPicture', () => {
  it('debe hacer mach con el snapshot', async () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 100,
        showPokemon: false
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('debe mostrar la imagen oculta y el pokemon 100', async () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 100,
        showPokemon: false
      }
    })

    const [img1, img2] = wrapper.findAll('img')
    expect(img1.exists()).toBe(true)
    expect(img1.classes('hidden-pokemon')).toBe(true)
    expect(img2).toBe(undefined)
    expect(img1.attributes('src')).toBe(
      'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg'
    )
  })

  it('debe mostrar el pokemon si showPokemon: true', async () => {
    const wrapper = shallowMount(PokemonPicture, {
      props: {
        pokemonId: 100,
        showPokemon: false
      }
    })

    const [img1, img2] = wrapper.findAll('img')
    expect(img1.exists()).toBe(true)
    expect(img1.classes('hidden-pokemon')).toBe(true)
    expect(img2).toBe(undefined)
  })
})
