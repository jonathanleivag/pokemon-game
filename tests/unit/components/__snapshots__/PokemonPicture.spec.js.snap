// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture debe hacer mach con el snapshot 1`] = `<div class="pokemon-container"><img class="hidden-pokemon" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg" alt="pokÃ©mon"></div>`;
